ethereum/core/src/main/java/org/hyperledger/besu/ethereum/forkid/ForkIdManager.java

```diff
@@ -162,21 +193,34 @@ private long createForkIds() {
     final CRC32 crc = new CRC32();
     crc.update(genesisHash.toArray());
     genesisHashCrc = getCurrentCrcHash(crc);
-    final List<Bytes> forkHashes = new ArrayList<>(List.of(genesisHashCrc));
-    forkBlockNumbers.forEach(
+    final List<Bytes> numberForkHashes = new ArrayList<>(List.of(genesisHashCrc));
+    blockNumberForks.forEach(
+        fork -> {
+          updateCrc(crc, fork);
+          numberForkHashes.add(getCurrentCrcHash(crc));
+        });
+
+    timestampForks.forEach(
         fork -> {
           updateCrc(crc, fork);
-          forkHashes.add(getCurrentCrcHash(crc));
+          numberForkHashes.add(getCurrentCrcHash(crc));
         });
 
     // This loop is for all the fork hashes that have an associated "next fork"
-    for (int i = 0; i < forkBlockNumbers.size(); i++) {
-      forkIds.add(new ForkId(forkHashes.get(i), forkBlockNumbers.get(i)));
+    for (int i = 0; i < blockNumberForks.size(); i++) {
+      blockNumbersForkIds.add(new ForkId(numberForkHashes.get(i), blockNumberForks.get(i)));
+    }
+    for (int i = 0; i < timestampForks.size(); i++) {
+      timestampsForkIds.add(
+          new ForkId(numberForkHashes.get(blockNumberForks.size() + i), timestampForks.get(i)));
     }
     long forkNext = 0;
-    if (!forkBlockNumbers.isEmpty()) {
-      forkNext = forkIds.get(forkIds.size() - 1).getNext();
-      forkIds.add(new ForkId(forkHashes.get(forkHashes.size() - 1), 0));
+    if (!timestampForks.isEmpty()) {
+      forkNext = timestampForks.get(timestampForks.size() - 1);
+      timestampsForkIds.add(new ForkId(numberForkHashes.get(numberForkHashes.size() - 1), 0));
+    } else if (!blockNumberForks.isEmpty()) {
```

Comment:
Why is this else if and not just if?

Reply:
The next forkId should be added to only one of the timestamp forks or the blockNumber forks. Using an if instead of the else if would break in the case where we had a non-empty blockNumberForks and non-empty timestampForks.